File: config/firebase.go
package config

import (
	"context"
	"fmt"
	"log"

	firebase "firebase.google.com/go"
	"firebase.google.com/go/messaging"
	"google.golang.org/api/option"
)

var FireBaseApp *firebase.App
var MessagingClient *messaging.Client

func InitFirebase() {

	opt := option.WithCredentialsFile("./firebase-config-be.json")

	// Firebase app
	app, err := firebase.NewApp(context.Background(), nil, opt)
	if err != nil {
		log.Fatalf("Firebase başlatma hatası: %v", err)
	}

	// Firebase Messaging Client
	client, err := app.Messaging(context.Background())
	if err != nil {
		log.Fatalf("Firebase Messaging başlatma hatası: %v", err)
	}

	FireBaseApp = app
	MessagingClient = client

	fmt.Println("🔥 Firebase başarıyla başlatıldı!")
}

File: filend-ui/src/utils/firebase.js
import { initializeApp } from "firebase/app";
import { getMessaging, getToken, onMessage } from "firebase/messaging";
import firebaseConfig from "@/config/firebaseConfig.json";

const firebaseApp = initializeApp(firebaseConfig);
const messaging = getMessaging(firebaseApp);

//console.log("Firebase initialized:", firebaseApp);
//console.log("Messaging initialized:", messaging);


export { messaging, getToken, onMessage };